docs/aws/audit/apigatewaymonitoring/rules/efs_kms_customer_master_keys_remediation.mdx

```diff
@@ -2,6 +2,196 @@
+</Accordion>
+
+<Accordion title='Using CLI'>
+To prevent the use of AWS KMS Customer Master Keys (CMKs) for EFS encryption in API Gateway using AWS CLI, you can follow these steps:
+
+1. **Create a Key Policy to Restrict Usage:**
+   Ensure that the key policy for your CMK restricts its usage to only the intended services and principals. This can be done by specifying the allowed actions and principals in the key policy.
+
+   ```sh
+   aws kms put-key-policy --key-id <your-key-id> --policy-name default --policy file://key-policy.json
+   ```
+
+   Example `key-policy.json`:
+   ```json
+   {
+     "Version": "2012-10-17",
+     "Id": "key-default-1",
+     "Statement": [
+       {
+         "Sid": "Allow administration of the key",
+         "Effect": "Allow",
+         "Principal": {
+           "AWS": "arn:aws:iam::111122223333:role/Admin"
+         },
+         "Action": [
+           "kms:*"
+         ],
+         "Resource": "*"
+       },
+       {
+         "Sid": "Allow use of the key",
+         "Effect": "Allow",
+         "Principal": {
+           "AWS": "arn:aws:iam::111122223333:role/EFSRole"
+         },
+         "Action": [
+           "kms:Encrypt",
+           "kms:Decrypt",
+           "kms:ReEncrypt*",
+           "kms:GenerateDataKey*",
+           "kms:DescribeKey"
+         ],
+         "Resource": "*"
+       }
+     ]
+   }
+   ```
+
+2. **Disable Key Rotation:**
+   Disable automatic key rotation for the CMK to ensure that the key is not automatically rotated, which might affect its usage.
+
+   ```sh
+   aws kms disable-key-rotation --key-id <your-key-id>
+   ```
+
+3. **Tagging the Key for Identification:**
+   Tag the CMK to identify it easily and ensure it is not used for unintended purposes.
+
+   ```sh
+   aws kms tag-resource --key-id <your-key-id> --tags TagKey=Purpose,TagValue=EFS-Only
+   ```
+
+4. **Audit Key Usage:**
+   Regularly audit the usage of the CMK to ensure it is only being used for the intended purpose. You can use AWS CloudTrail to monitor the usage of the key.
+
+   ```sh
+   aws cloudtrail lookup-events --lookup-attributes AttributeKey=ResourceName,AttributeValue=<your-key-id>
+   ```
+
+By following these steps, you can prevent the misuse of AWS KMS Customer Master Keys for EFS encryption in API Gateway using AWS CLI.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent AWS KMS Customer Master Keys (CMKs) for EFS (Elastic File System) encryption in API Gateway using Python scripts, you can follow these steps:
+
+### Step 1: Install Boto3
+Ensure you have the Boto3 library installed, which is the AWS SDK for Python.
+
+```bash
+pip install boto3
+```
+
+### Step 2: Create a Boto3 Session
+Create a Boto3 session to interact with AWS services.
+
+```python
+import boto3
+
+# Create a session using your AWS credentials
+session = boto3.Session(
+    aws_access_key_id='YOUR_AWS_ACCESS_KEY_ID',
+    aws_secret_access_key='YOUR_AWS_SECRET_ACCESS_KEY',
+    region_name='YOUR_AWS_REGION'
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>